src/main/java/cow/farm/utils/LukaszGenericArrayList.java

```diff
@@ -180,15 +184,14 @@ public E set(int index, E element) {
 	}
 
 	@Override
-	public List subList(int fromIndex, int toIndex) {
-		if(fromIndex>=0 & toIndex>=0){
-			E[] subArray = (E[]) new Object[toIndex-fromIndex];
-			System.arraycopy(array, fromIndex, subArray, 0, toIndex-fromIndex);
-			//FIXME finish me
-			//return subArray;
-		}
-		return null;
-	}
+    //FIXME why return is red !!!!!!!!!!!!!!!! ????????????
+    //TODO call Jacek to fix or die....trying
+    public List<E> subList(int fromIndex, int toIndex) {
+        checkIndexBounds(fromIndex);
+        checkIndexBounds(toIndex);
+        E subArray = (E) Arrays.copyOfRange(array, fromIndex, toIndex);
```

Comment:
ok, to co chcesz zrobić tutaj to wybrać częściowo tablicę. Zwracana z funkcji tablica jest typu E[], nie E. Co więcej, Ty  chcesz zwrócić Listę[E], nie tablicę E[]. Tak więc przed zwróceniem stwórz tablicę (LukaszGenericArrayList<E>), podając w konstruktorze tablicę.